scripts/new-version-pr-generator.js

```diff
@@ -90,57 +90,65 @@ The automatic branch creation for ${repository} failed. This can be due to many
-        {
-          cwd: sandbox,
-          stdio: 'inherit',
-        },
-      ),
-    );
-
-    const commands = [
-      `git checkout -b update-polaris-${releaseVersion}`,
-      `npx yarn@${YARN_VERSION} upgrade @shopify/polaris@${releaseVersion.replace(
-        'v',
-        '',
-      )} --no-progress --ignore-engines`,
-      `git add package.json yarn.lock`,
-      `${shopifyPolarisBotGitOverride} git commit -m "Update @shopify/polaris to ${releaseVersion}" --author "${polarisBotName} <${polarisBotEmail}>" -m "${releaseVersion}" --allow-empty`,
-      `git push origin update-polaris-${releaseVersion}`,
-      `curl -d '${JSON.stringify({
-        title: `Update @shopify/polaris to ${releaseVersion} 🚀`,
-        body: pullRequestTemplate(repository, releaseVersion),
-        head: `update-polaris-${releaseVersion}`,
-        base: baseBranch,
-      })}' -X POST https://api.github.com/repos/shopify/${repository}/pulls?access_token=${polarisBotToken}`,
-    ];
-
-    // Run the commands in the cloned repository directories
-    commands.forEach((command) => {
-      retry(
-        execSync(command, {
-          cwd: repositoryDirectory,
-          stdio: 'inherit',
-        }),
+const jobs = repositories.map((repository) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const repositoryDirectory = path.resolve(sandbox, repository);
+
+      // Clone into the sandbox directory
+      await retry(() => {
+        execSync(
+          `git clone --branch ${baseBranch} --single-branch https://${polarisBotToken}@github.com/Shopify/${repository}.git`,
+          {
+            cwd: sandbox,
+            stdio: 'inherit',
+          },
+        );
+      });
+
+      const commands = [
+        `git checkout -b update-polaris-${releaseVersion}`,
+        `npx yarn@${YARN_VERSION} upgrade @shopify/polaris@${releaseVersion.replace(
+          'v',
+          '',
+        )} --no-progress --ignore-engines`,
+        `git add package.json yarn.lock`,
+        `${shopifyPolarisBotGitOverride} git commit -m "Update @shopify/polaris to ${releaseVersion}" --author "${polarisBotName} <${polarisBotEmail}>" -m "${releaseVersion}" --allow-empty`,
+        `git push origin update-polaris-${releaseVersion}`,
+        `curl -d '${JSON.stringify({
+          title: `Update @shopify/polaris to ${releaseVersion} 🚀`,
+          body: pullRequestTemplate(repository, releaseVersion),
+          head: `update-polaris-${releaseVersion}`,
+          base: baseBranch,
+        })}' -X POST https://api.github.com/repos/shopify/${repository}/pulls?access_token=${polarisBotToken}`,
+      ];
+
+      // Run the commands in the cloned repository directories
+      commands.forEach(async (command) => {
+        await retry(() => {
+          execSync(command, {
+            cwd: repositoryDirectory,
+            stdio: 'inherit',
+          });
+        });
+      });
```

Comment:
I think this has has the same problem we talked about before, these commands are executed in parallel not sequentially. Try replacing the retry with a sleep like we we tried Friday.

For this to be executed sequentially you **can not use an array abstraction**. All array abstractions are going to be executed in parallel.

Either chain promises or use execSync sequentially, either works.
```js
execSync(`git checkout -b update-polaris-${releaseVersion}`, {
  cwd: repositoryDirectory,
  stdio: 'inherit',
});
execSync(`npx yarn@${YARN_VERSION} upgrade @shopify/polaris@${releaseVersion.replace(
  'v',
  '',
)} --no-progress --ignore-engines`, {
  cwd: repositoryDirectory,
  stdio: 'inherit',
});
```

Reply:
I have fixed this up, let me know if you find the solution ok!